frontend/test/validate-e2e-test-files.js

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env node
+const { execSync } = require("child_process");
+
+const path = require("path");
+const glob = require("glob");
+const chalk = require("chalk");
+
+const E2E_FILE_EXTENSION = ".cy.spec.js";
+const E2E_HOME = "frontend/test/metabase/scenarios/";
+
+init();
+
+function validateE2EFileNames(files) {
+  if (!files || !Array.isArray(files)) {
+    return;
+  }
+
+  const invalidFileNames = files.filter(fullPath => {
+    return !path.basename(fullPath).endsWith(E2E_FILE_EXTENSION);
+  });
+
+  printFeedback(invalidFileNames);
+}
+
+/**
+ * @param {Array<string>} [files] Potential payload if used with `lint-staged`
+ */
+function validateStagedFiles(files) {
+  const stagedFiles = getStagedFiles();
+  validateE2EFileNames(files || stagedFiles);
```

Comment:
nit: so we don't have to get the `stagedFiles` every time despite `files` being passed. If we use this it'll short circuit at `files` then `getStagedFiles()` will never be invoked when passing `files`.
```suggestion
  validateE2EFileNames(files || getStagedFiles());
```